File: December 2022/02 December 2022/C++/Check if it is possible to convert one string into another with given constraints.cpp
//{ Driver Code Starts
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// User function Template for C++

class Solution {
  public:
    int isItPossible(string S, string T, int M, int N) {
        // code here
        
        int i=0, j=0;
        
        if(M!=N) return 0;
        
        while(i<M || j<N){
            
            while(S[i]=='#') i++;
            while(T[j]=='#') j++;
            
            if(i>=M && j<N) return 0;
            if(i<M && j>=N) return 0;
            
            if(S[i]!=T[j]) return 0;
            else{
                if(S[i]=='A' && i<j) return 0;
                else if(S[i]=='B' && i>j) return 0;
            }
            
            i++;
            j++;
        }
        
        return 1;
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        string S, T;
        cin >> S >> T;
        int M = S.length(), N = T.length();
        Solution ob;
        cout << ob.isItPossible(S, T, M, N);
        cout << "\n";
    }
}
// } Driver Code Ends

File: Problems & Solutions/C/Check if it is possible to convert one string into another with given constraints/02 December 2022/Python/Check if it is possible to convert one string into another with given constraints.py
#User function Template for python3

class Solution:
    def isItPossible(sef, S, T, M, N):
        i=0
        j=0
        if M!=N:
            return 0
        while i<M or j<N:
            while i<M and S[i]=="#":
                i+=1
            while j<N and T[j]=="#":
                j+=1
            if i>=M and j<N:
                return 0
            if i<M and j>=N:
                return 0
            if i>=M and j>=N:
                return 1
            if S[i]!=T[j]:
                return 0
            else:
                if S[i]=="A" and i<j:
                    return 0
                elif S[i]=="B" and i>j:
                    return 0
            i+=1
            j+=1
        return 1

#{ 
 # Driver Code Starts
#Initial Template for Python 3

import math
        
if __name__=='__main__':
    t=int(input())
    for _ in range(t):
        S,T=input().split()
        ob=Solution()
        print(ob.isItPossible(S,T,len(S),len(T)))
# } Driver Code Ends